### ISSUE TITLE
reduxForm.values does not contain unchecked checkbox values

### ISSUE THREAD

**Issue by binarykitchen** [erikras/redux-form#3617]
When I have checkboxes in my redux form and do not have them checked, `reduxForm.values` doesn't include these. Which is a problem when you want to pass on that further to i.E. API for posting to a server which then complains about the missing field.

Happens on latest version 7.1.2 - I would include unchecked checkboxes and set them just to false.

As a temporary workaround I am thinking of looping through `registeredFields` instead which is kind ugly.

**Comment by gustavohenke** [comment]
Hi @binarykitchen,
Redux form doesn't initialise any form values. This is up to your implementation.

A few suggestions:
- Initialise your form passing the missing fields as whatever makes sense
- Wrap your elements and initialise their value internally somehow (eg dispatch an action that changes itself)

**Comment by binarykitchen** [comment]
Yes, this workaround helps but is pretty ugly. Using this with immutable altogether for something like `Api.post(... values())`

```js
  values () {
    const registeredFields = this.props.someForm.get('registeredFields')
    const values = this.props.someForm.get('values')
    // reason for this is to set `false` for the private checkbox when it's
    // not in the map of values (weird redux-form behavior)
    return registeredFields.map((v, k) => values.get(k, false)) // here I am setting false for missing values coming from checkbox fields
  }
```

**Comment by binarykitchen** [comment]
@gustavohenke I doubt that "initialising" will help. What if the user ticks and unticks the checkbox multiple times over again?

**Comment by gustavohenke** [comment]
Oh well, now I'm completely lost. This must have something to do with me not using Redux Form in a good while.

https://codesandbox.io/s/6y0270qq3r

**Comment by binarykitchen** [comment]
Ping?
